Killing idle backend failure.

<!-- Thanks for helping us to improve LocalAI! We welcome all bug reports. Please fill out each area of the template so we can better help you. Comments like this will be hidden when you post but you can delete them if you wish. -->

**LocalAI version:**
sha-ff88c39-cublas-cuda12-ffmpeg (docker image)

**Environment, CPU architecture, OS, and Version:**
(Standard docker image)
Linux LocalAi 6.5.11-8-pve #1 SMP PREEMPT_DYNAMIC PMX 6.5.11-8 (2024-01-30T12:27Z) x86_64 x86_64 x86_64 GNU/Linux

**Describe the bug**
Killing idle llama backend from watchdog causes invalid pointer crash.

**To Reproduce**
Set enviorment variables:
```
SINGLE_ACTIVE_BACKEND=true
BUILD_TYPE=cublas
REBUILD=true
BUILD_GRPC_FOR_BACKEND_LLAMA=true
PYTHON_GRPC_MAX_WORKERS=3
LLAMACPP_PARALLEL=1
PARALLEL_REQUESTS=true
WATCHDOG_IDLE=true
WATCHDOG_IDLE_TIMEOUT=5m
```

Load multiple (2+) models with llama in succession then let it idle allowing watchdog to kill the backend.

**Expected behavior**
Backend should shutdown gracefully.

**Logs**
11:41PM WRN [WatchDog] Address 127.0.0.1:35825 is idle for too long, killing it
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xb95344]

goroutine 39 [running]:
github.com/mudler/go-processmanager.(*Process).path(...)
        /root/go/pkg/mod/github.com/mudler/go-processmanager@v0.0.0-20230818213616-f204007f963c/process.go:32
github.com/mudler/go-processmanager.(*Process).readPID(0x0)
        /root/go/pkg/mod/github.com/mudler/go-processmanager@v0.0.0-20230818213616-f204007f963c/process.go:52 +0x24
github.com/mudler/go-processmanager.(*Process).Stop(0x0)
        /root/go/pkg/mod/github.com/mudler/go-processmanager@v0.0.0-20230818213616-f204007f963c/process.go:150 +0x1c
github.com/go-skynet/LocalAI/pkg/model.(*ModelLoader).deleteProcess(0xc0004dee00, {0xc000a3a200, 0x20})
        /build/pkg/model/process.go:32 +0x45
github.com/go-skynet/LocalAI/pkg/model.(*ModelLoader).StopModel(0xd1f720?, {0xc000a3a200, 0x20})
        /build/pkg/model/loader.go:160 +0x111
github.com/go-skynet/LocalAI/pkg/model.(*WatchDog).checkIdle(0xc0005d4240)
        /build/pkg/model/watchdog.go:115 +0x2c2
github.com/go-skynet/LocalAI/pkg/model.(*WatchDog).Run(0xc0005d4240)
        /build/pkg/model/watchdog.go:99 +0xdd
created by github.com/go-skynet/LocalAI/core/http.Startup in goroutine 1
        /build/core/http/api.go:100 +0x8fe

**Additional context**
This only tends to happen when multiple models are called, when it is only a single model loaded regardless of requests sent the crash does not occur.

--- a/pkg/model/loader.go
+++ b/pkg/model/loader.go
@@ -155,10 +155,10 @@ func (ml *ModelLoader) ShutdownModel(modelName string) error {
 	ml.mu.Lock()
 	defer ml.mu.Unlock()
 
-	return ml.StopModel(modelName)
+	return ml.stopModel(modelName)
 }
 
-func (ml *ModelLoader) StopModel(modelName string) error {
+func (ml *ModelLoader) stopModel(modelName string) error {
 	defer ml.deleteProcess(modelName)
 	if _, ok := ml.models[modelName]; !ok {
 		return fmt.Errorf("model %s not found", modelName)

--- a/pkg/model/watchdog.go
+++ b/pkg/model/watchdog.go
@@ -30,7 +30,7 @@ type WatchDog struct {
 }
 
 type ProcessManager interface {
-	StopModel(modelName string) error
+	ShutdownModel(modelName string) error
 }
 
 func NewWatchDog(pm ProcessManager, timeoutBusy, timeoutIdle time.Duration, busy, idle bool) *WatchDog {
@@ -112,9 +112,10 @@ func (wd *WatchDog) checkIdle() {
 			log.Warn().Msgf("[WatchDog] Address %s is idle for too long, killing it", address)
 			p, ok := wd.addressModelMap[address]
 			if ok {
-				if err := wd.pm.StopModel(p); err != nil {
+				if err := wd.pm.ShutdownModel(p); err != nil {
 					log.Error().Msgf("[watchdog] Error shutting down model %s: %v", p, err)
 				}
+				log.Debug().Msgf("[WatchDog] model shut down: %s", address)
 				delete(wd.idleTime, address)
 				delete(wd.addressModelMap, address)
 				delete(wd.addressMap, address)
@@ -139,9 +140,10 @@ func (wd *WatchDog) checkBusy() {
 			model, ok := wd.addressModelMap[address]
 			if ok {
 				log.Warn().Msgf("[WatchDog] Model %s is busy for too long, killing it", model)
-				if err := wd.pm.StopModel(model); err != nil {
+				if err := wd.pm.ShutdownModel(model); err != nil {
 					log.Error().Msgf("[watchdog] Error shutting down model %s: %v", model, err)
 				}
+				log.Debug().Msgf("[WatchDog] model shut down: %s", address)
 				delete(wd.timetable, address)
 				delete(wd.addressModelMap, address)
 				delete(wd.addressMap, address)
@@ -149,7 +151,6 @@ func (wd *WatchDog) checkBusy() {
 				log.Warn().Msgf("[WatchDog] Address %s unresolvable", address)
 				delete(wd.timetable, address)
 			}
-
 		}
 	}
 }

